examples/recording_service/leveldb-integration/cpp/README.md

```diff
@@ -0,0 +1,1077 @@
+  instance to create an instance of this class during startup, 
+  by calling method `create_stream_info_reader()`.
+  The instance of this class will be in charge of reading from the 
+  `DCPSPublication.dat` database.
+  The main methods this class implements are the `read()` and
+  `return_loan()` methods.
+  It also implements two methods that *Replay* uses to obtain the
+  database's start and end times, `service_start_time()` and
+  `service_stop_time()`.
+  It also implements the `fisnished()` method to tell *Replay* or
+  *Converter* when there are no more discovery samples to read.
+  The `reset()` method is used by *Replay* when the looping
+  option is enabled.
+  
+- **`LevelDbStreamReader`**: this class extends the API class
+  `rti::recording::storage::DynamicDataStorageStreamReader`. This API
+  class is a specialization of class `StorageStreamReader` set up to
+  work with Dynamic Data samples. When a user-data topic is discovered
+  (read by Replay using the stream info reader above) and matches the
+  filters in the *Replay* or *Converter* configuration, the application 
+  will ask the `LevelDbReader` instance to create an instance of this 
+  class, by calling method `create_stream_reader()`.
+  The instances of this class will look for and read the appropriate 
+  `<topic-name>@<domain-id>` database where samples of the topic are 
+  stored. 
+  The main method this class has to implement are `read()` and 
+  `return_loan()`. This methods will be called by the application when 
+  it's time to read new samples from the database. The `return_loan()`
+  operation is called by the application after the read samples have
+  been used so that resources associated with the samples can be freed. 
+  It also implements the `finished()` method to tell *Replay* or
+  *Converter* when there are no more samples for the topic to read.
+  The `reset()` method is used by *Replay* when the looping option 
+  is enabled.
+  
+#### Implementation details of class `LevelDbStreamReader`
+
+This Storage Reader plug-in works by reading data stored in 
+_serialized_ CDR format, by using the types described in the above section
+([Format of the stored data](#format-of-the-stored-data)). The code
+for the types defined in file `LevelDb_RecorderTypes.idl` will be
+generated automatically. The generated code includes methods to
+serialize and deserialize samples to and from CDR format.
+
+The main members of the `LevelDbStreamReader` class are:
+
+- `data_db_`: this is the LevelDB database object (`leveldb::DB`) that 
+  will be used to read the data and that was recorded by *Recorder* using
+  the Storage Writer implementation in this example.
+- `db_iterator_`: the class keeps a database iterator (`leveldb::Iterator`)
+  that points to the next sample to be read.
+- `current_timestamp_`: the current sample's reception timestamp.
+- `key_comparator`: the custom key comparator will be passed when
```

Comment:
this seems to be a copy-carry error (assuming it is)

Reply:
I just corrected all occurrences.